siriuspy/siriuspy/orbintlk/main.py

```diff
@@ -1358,13 +1358,23 @@ def _callback_bpm_intlk(self, pvname, value, **kws):
         if not value:
             return
         # launch thread to log interlock details
+        bpmname = _PVName(pvname).device_name
         _CAThread(
             target=self._log_bpm_intlk,
-            args=(_PVName(pvname).device_name, ),
+            args=(bpmname, ),
             daemon=True).start()
         # launch thread to send interlock to RF as a backup
         if self._thread_cbbpm and self._thread_cbbpm.is_alive():
             return
+
+        # NOTE: the next lines help to avoid killing beam in case one BPM that
+        # is not enabled raises a false positive interlock signal.
+        idx = self._const.bpm_idcs[bpmname]
```

Comment:
Could we avoid having to define `bpm_idcs` and just do `self._const.bpm_names.index(bpmname)` instead?

Reply:
The method `list.index` is approximately 50x slower than checking a dictionary and we want to send the manual interlock as fast as possible.

Reply:
That's fair, I hadn't considered that this is a critical path.